### ISSUE TITLE
Debugger fails to attach when launching the Debugger for App Engine Dev Server via gradle appengineRun

### ISSUE THREAD

**Issue by jamesmortensen** [GoogleCloudPlatform/cloud-code-intellij#2936]
### Expected Behavior

I followed the instructions to [Setup the Debugger for Google App Engine Standard with Gradle and Intellij Community Edition](https://cloud.google.com/code/docs/intellij/deploy-local#debugging_your_application_locally_on_community_edition)

When running with the debugger in Intellij, I expect the execution to stop on a breakpoint.

### Actual Behavior

When running the application, none of the breakpoints are stopping the execution even though I can see the logs and verify the code is executing.

**Screenshot of setting up run configuration for the Gradle Task:**
<img width="957" alt="Screen Shot 2021-05-29 at 10 07 31 PM" src="https://user-images.githubusercontent.com/1315816/120077949-945d4780-c0ca-11eb-83bc-59758a1f48bc.png">

**Screenshot of the Remote Debugging Configuration:**
<img width="950" alt="Screen Shot 2021-05-29 at 10 08 40 PM" src="https://user-images.githubusercontent.com/1315816/120077963-a7701780-c0ca-11eb-94e5-e65902e130e9.png">

Section of the build.gradle where I configured app engine with the jvmArgs:

```
appengine {
  run {
    host = "0.0.0.0"
    jvmFlags = [
            "-Ddatastore.backing_store=/Users/jem/.gcp_data/SetMore/local_db.bin",
            '-Xdebug', '-Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=5009'
    ]
    automaticRestart = true
  }
```

I originally tried with port 5005, but with it not working, I thought I would try with different ports.  Even with 5009, it doesn't connect.

What's interesting is that, when executing the remote debug configuration, the log says it is connecting to a random port.  Shouldn't it say it's connecting to port 5005 or 5009?

**In the log below, note the port 50303.  Where did this come from?**
<img width="1011" alt="Screen Shot 2021-05-29 at 10 15 09 PM" src="https://user-images.githubusercontent.com/1315816/120078091-5e6c9300-c0cb-11eb-8f6f-a8cc49824c0e.png">

### Additional Information

- IDE type  IntelliJ Community Edition
- IDE version: 2021.1.1 built on April 30, 2021
- Cloud Code version:  21.5.1
- Cloud SDK:  342
  - Are you allowing the plugin to manage the Cloud SDK:   I don't think so.
  - Version of the Cloud SDK:   342
- Operating System:  Mac OS 10.15.1
- gradle 4.10.3.

I also tried with both Oracle Java 8:

```
java version "1.8.0_162"
Java(TM) SE Runtime Environment (build 1.8.0_162-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.162-b12, mixed mode)
```

As well as AdoptOpenJDK 8:

```
openjdk version "1.8.0_292"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_292-b10)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.292-b10, mixed mode)
```

The results are the same.  

I'm out of ideas on getting the debugger working and can only think that it's possibly some kind of bug.  If there is any other information I can provide, please let me know as I think it's important that developers can step through code with a debugger in order to understand what it's doing.  Thank you.

**Comment by jamesmortensen** [comment]
Here is a follow up on this issue.  I decided to try a different port, other than 5005.  I changed both the build.gradle appengine run jvmFlag port to 5009, and I also updated the port to 5009 in the Remote Debugging configuration.  After this change and a computer reboot, I noticed in the console log that there was a message saying `Listening for transport dt_socket at address: 5009` followed by a hyperlink "Attach Debugger".

I also had missed seeing this link.  All the documentation on the debugging in IntelliJ with Gradle suggests the process auto-connects, but I needed to click "Attach Debugger".

Once I clicked this link, I can now load the http://localhost:8080 page and see the breakpoints get hit and inspect variables.

In summary, I faced two major issues:

- A blocked port 5005.
- Unclear documentation on needing to click "Attach Debugger".

One suggestion, and I'm not sure where this should be documented, here or elsewhere, is if there could be better error handling or error messages when ports are blocked.  

We can also close this issue as the issue was not with IntelliJ or cloud code but due to a blocked port.  Hope this helps others struggling to set up the debugger in IntelliJ with Google App Engine.

**Comment by etanshaul** [comment]
@jamesmortensen thanks for the feedback, and for sharing the solution that worked.

cc @frostie to help take a look at the docs here to make things clearer.

I'm going to close the issue for now, but feel free to follow up with any questions.
